### PR TITLE
Add basic documentation for amp-lightbox-gallery

### PR DESCRIPTION
**Pull request by cathyxz**
Basic documentation for an experimental component `<amp-lightbox-gallery>`. The API for styling, thumbnails, and supported elements is still in progress, hence there is only basic usage under an experiment flag.

### CODE REVIEW

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  <tr>
+    <td class="col-fourty"><strong><a href="https://www.ampproject.org/docs/guides/responsive/control_layout.html">Supported Layouts</a></strong></td>
+    <td>nodisplay</td>
+  </tr>
+</table>
+
+## Behavior
+All you need to do is to include the 'lightbox' attribute on an `<amp-img>` or `<amp-carousel>`. A typical usage looks like this:
+
+### Lightbox with AMP-IMG
```
Comment (aghassemi) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: lowecase amp-img

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <td><code>&lt;script async custom-element="amp-lightbox-gallery" src="https://cdn.ampproject.org/v0/amp-lightbox-gallery-0.1.js">&lt;/script></code></td>
+  </tr>
+  <tr>
+    <td class="col-fourty"><strong><a href="https://www.ampproject.org/docs/guides/responsive/control_layout.html">Supported Layouts</a></strong></td>
+    <td>nodisplay</td>
+  </tr>
+</table>
+
+## Behavior
+All you need to do is to include the 'lightbox' attribute on an `<amp-img>` or `<amp-carousel>`. A typical usage looks like this:
```
Comment (aghassemi) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: ...is to include the extension script and put the `lightbox` attribute...

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+### Lightbox with AMP-IMG
+
+```html
+<amp-img src="image1" width=200 height=100 lightbox></amp-img>
+<amp-img src="image2" width=200 height=100 lightbox></amp-img>
+```
+
+The `<amp-lightbox-gallery>` extension automatically inserts an invisible `<amp-lightbox-gallery>` component (with the id 'amp-lightbox-gallery') into the dom.
```
Comment (aghassemi) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: I don't think this is needed, it is very internal implementation details.

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+### Lightbox with AMP-CAROUSEL
+```html
+<amp-carousel lightbox width=1600 height=900 layout=responsive type=slides>
+  <amp-img src="image1" width=200 height=100></amp-img>
+  <amp-img src="image2" width=200 height=100></amp-img>
+  <amp-img src="image3" width=200 height=100></amp-img>
+</amp-carousel>
+```
+
+`<amp-carousel>` is considered a lightbox container item. To process an `<amp-carousel>` with the `lightbox` attribute,  `<amp-lightbox-gallery>` extension will iterate all of the `<amp-carousel>`'s children, add the `lightbox` attribute and a tap-handler to each child.
```
Comment (aghassemi) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: let's mention the carousel is kept in sync with lightbox gallery

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+### Captions
+Lightbox optionally allows the user to specify a caption for each element. These fields are automatically read and displayed by `<amp-lightbox-gallery>` in the following order of priority:
+- figcaption (if the lightboxed element is a figure)
```
Comment (aghassemi) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: `if lightboxed element is a child of a figure`

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md (hunk 3 above)
Comment (bpaduch) [reply extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: Agreed.

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+### Captions
+
+Optionally, you can specify a caption for each element in the lightbox. These fields are automatically read and displayed by the `<amp-lightbox-gallery>` in the following order of priority:
+
+- `figcaption` (if the lightboxed element is the child of a figure)
+- `aria-describedby`
+- `alt`
+- `aria-label`
+- `aria-labelledby`
+
```
Comment (bpaduch) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: Suggest a code snippet example

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  </tr>
+  <tr>
+    <td class="col-fourty"><strong><a href="https://www.ampproject.org/docs/guides/responsive/control_layout.html">Supported Layouts</a></strong></td>
+    <td>nodisplay</td>
+  </tr>
+</table>
+
+## Overview
+
+The `amp-lightbox-gallery` component provides a "lightbox” experience for AMP components (e.g., `amp-img`, `amp-carousel`). When the user interacts with the AMP element, a UI component expands to fill the viewport until it is closed by the user. Currently, only images are supported.
```
Comment (bpaduch) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: I reworded this.  Is it only for AMP components or can it be applied to HTML elements too?

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 8]
```diff
@@ -37,9 +37,9 @@ limitations under the License.
...
 
 ## Behavior
-All you need to do is to include the 'lightbox' attribute on an `<amp-img>` or `<amp-carousel>`. A typical usage looks like this:
+All you need to do is to include the extension script and put the 'lightbox' attribute on an `<amp-img>` or `<amp-carousel>`. A typical usage looks like this:
 
-### Lightbox with AMP-IMG
+### Lightbox with `<amp-img>`
 
 ```html
 <amp-img src="image1" width=200 height=100 lightbox></amp-img>
```
Comment (bpaduch) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: In code samples, attribute values are enclosed in double quotes (e.g., `"width="200"`). Please apply that standard to your code sample here and below.

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+The `amp-lightbox-gallery` component provides a "lightbox” experience for AMP components (e.g., `amp-img`, `amp-carousel`). When the user interacts with the AMP element, a UI component expands to fill the viewport until it is closed by the user. Currently, only images are supported. 
+
+## Usage
+
+To use `amp-lightbox-gallery`, ensure the required script is included in your `<head>` section, then add the `lightbox` attribute on an `<amp-img>` or `<amp-carousel>` element. A typical usage looks like this:
+
+### Lightbox with `<amp-img>`
+
+```html
+<amp-img src="image1" width=200 height=100 lightbox></amp-img>
```
Comment (bpaduch) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: In code samples, attribute values are enclosed in double quotes (e.g., `"width="200"`). Please apply that standard to your code sample here and below.

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md (hunk 7 above)
Comment (bpaduch) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: I reworded this.  Is it only for AMP components or can it be applied to HTML elements too?

Comment (cathyxz) [reply extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: I think we're not quite sure yet. We're currently supporting things on a whitelisted basis, which only includes `amp-img` and `amp-carousel`. We should be able to support divs, but my understanding is that AMP doesn't really allow very many raw html elements? E.g.  and <video> aren't allowed. Realistically are there html elements that make sense to support other than divs? 

I'm open to changing the wording on this so long as we make our support guarantees explicit until we're ready.

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md (hunk 6 above)
Comment (cathyxz) [reply extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: Done.

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md (hunk 3 above)
Comment (cathyxz) [reply extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: Done

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md (hunk 9 above)
Comment (cathyxz) [reply extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: Done.

**Pull request merged**